### DISCUSSION TITLE
[ZK Stack] - How to deploy the WETH on layer2?

### DISCUSSION THREAD

**Discussion by SuccinctPaul** [zkSync-Community-Hub/zksync-developers#642] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [x] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.78.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

* In the latest code (release-prover-16.0.0), after init and start sequencer, the `WETH` didn't deployed. This is for why?
* how to deploy the WETH(Wrapped BaseToken) on l2? As the older code has `deployL2Weth` method to do it. https://github.com/matter-labs/era-contracts/blob/d85a73a1eeb5557343b7b44c6543aaf391d8b984/l2-contracts/src/deployL2Weth.ts 

### Expected Behavior

* works fine

### Current Behavior

* doesn't work

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To deploy WETH on ZKsync Layer 2, you need to follow the process outlined in the ZKsync documentation. Here’s a brief overview:

1. **Deploy from L1**: The WETH contract is deployed from the Ethereum mainnet (L1) to ZKsync Layer 2. This involves using the L1 WETH bridge, which handles the deployment and initialization of the corresponding L2 WETH contract.

2. **Initialization**: The L1 WETH bridge deploys the L2 WETH contract. The address for the L2 contract is known at the time of deployment.

3. **Using the Bridge**: To deposit WETH into ZKsync L2, use the WETH bridge on L1. Specify the destination chain ID during this process. The bridge unwraps the WETH to ETH on L1, sends a message to L2, and mints WETH on L2 to the specified address.

For a visual guide on how the WETH is deployed and how deposits are handled, you can refer to the following images from the ZKsync documentation:
- [Deploy WETH Process](https://helixuserimages.s3.amazonaws.com/zksync_dev/c547f9e5-4987-4f51-8356-6a252f83cada.jpg)
- [Deposit WETH Process](https://helixuserimages.s3.amazonaws.com/zksync_dev/80f1b942-8bb3-467c-9dee-41303fac2ad4.jpg)

For more detailed instructions and technical specifics, please visit the [Shared Bridges section of the ZKsync documentation](https://docs.zksync.io/zk-stack/components/shared-bridges#components-weth-contract).

If you encounter issues with the deployment process or if the behavior differs from what's expected, ensure that your setup matches the requirements and versions specified in the ZKsync documentation.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply]
Were you able to get WETH deployed?

**Reply by SuccinctPaul** [reply]
When I executed `zk init` and `zk server`, WETH weren't deployed.

**Reply by bxpana** [reply, marked as answer]
It seems that WETH is not currently supported on ZK Chains, but something that he team is working on for the future. You could still deploy WETH as a simple contract as it is on Era, but for now it does not come default.
